File: 프로그래머스/LV2/코딩테스트 연습/연습문제/연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <vector>
#include <set>
using namespace std;

int solution(vector<int> elements) {
    set<int> answer;

    int elements_len = elements.size();
    vector<int> sums(elements_len, 0);

    for(int i = 0; i < elements_len - 1; i++) {
        for(int j = 0; j < elements_len; j++)
            sums[j] += elements[(i+j) % elements_len];
        
        for(int s : sums)
            answer.insert(s);
    }

    return answer.size() + 1;
}

File: 프로그래머스/LV2/코딩테스트 연습/연습문제/연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = set()

    elements_len = len(elements)
    sums = [0] * elements_len

    for i in range(elements_len-1):
        for j in range(elements_len):
            sums[j] += elements[(i+j) % elements_len]

        for s in sums:
            answer.add(s)

    return len(answer) + 1


